### PR TITLE
Bl 902 bd migrate and seed courses table

### PR DESCRIPTION
**Pull request by razborovdb**
## Description

1. Added to schema.sql SQL queries to migrate tables: admins,courses, instructors, instructors_program_types, programs
2. Added to data.sql SQL queries to seed tables: admins,courses, instructors, instructors_program_types, programs
3. Created a new package “hibernate” in main\java\com.bloomtechlabs.coderheroesbea  
4. Created class PostgreSqlStringArrayType that allows to get access to text[] column in Courses table,
   because Hibernate doesn't support this type for PostgreSQL
5. Created class Admins for table admins
6. Created class Courses for table courses
7. Created class Instructors for table instructors
8. Created class Instructors_program_types for table instructors_program_types
9. Created class Programs for table programs
10. Created interface AdminsRepository to access data in table admins
11. Created interface CoursesRepository to access data in table courses
12. Created interface InstructorsRepository to access data in table instructors
13. Created interface Instructors_program_typesRepository to access data in table instructors_program_types
14. Created interface ProgramsRepository to access data in table programs
15. Created test classes MigrationSeedAdminsTest, MigrationSeedCoursesTest, MigrationSeedInstructorsTest, 
MigrationSeedInstructors_program_typesTest, MigrationSeedProgramsTest for test migration and seed 
for tables: admins,courses, instructors, instructors_program_types, programs.

## Loom Video

https://www.loom.com/share/19263c5cdb234383bf28ecb7d0d6db0e

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [ ] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by decagondev — DISMISSED** [review]
The use of hibernate to convert the array type in your code was a good use of the library. Your entities follow the schema well and your code is clear and concise. You have consistently worked to a good standard with clean and clear code that is a pleasure to read.  The tests are written well and suit the purpose perfectly. This is a good example of what to do in the given ticket set.

File: src/main/java/com/bloomtechlabs/coderheroesbea/entities/Courses.java [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+import javax.persistence.*;
+import java.sql.Time;
+import java.util.Date;
+import java.util.List;
+
+@Entity
+@Table(name = "courses")
+
+//@Typ
```
Comment (tkozzer) [inline src/main/java/com/bloomtechlabs/coderheroesbea/entities/Courses.java]: Consider removing these dead code comments above the class definition

**Review by tkozzer — DISMISSED** [review]
Everything is clear and the code looks really clean. There was a bit of deadcode in your Courses.java file that you could probably delete (or put notes by if it's necessary). The only one thing that kind of stuck out to me is the use of snake case instead of camel case for one of your file names and inside many of your java classes. I was trying to look in the lab docs, and I couldn't find anything about naming conventions/styles, so ignore this if it's allowed. 

Otherwise, looks good to me.

File: src/main/java/com/bloomtechlabs/coderheroesbea/entities/Courses.java (hunk 1 above)
Comment (razborovdb) [reply src/main/java/com/bloomtechlabs/coderheroesbea/entities/Courses.java]: Remove dead code comments

**Review by tkozzer — APPROVED** [review]
Looks good Dmitry!

**Review by decagondev — APPROVED** [review]
looks like you have done the requested updates as per Tylers review. Looks good so far

**Pull request merged**